### PR TITLE
Fi-3467 add verifies requirements infrastructure

### PR DESCRIPTION
**Pull request by carrils**
# Summary
Adds inferno runnable for verifies_requirement, /inferno_requirement_tools/, and template file/documentation for  `requirements_config.yaml`

# Testing Guidance
Create mock attestation test files that contain `verifies_requirements` and reference requirements in the bulk data v2 requirements Excel.

### CODE REVIEW

File: lib/template_requirements_config.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+test_kit_id: # the name of the repository slug. E.g., onc-certification-g31-test-kit
```
Comment (karlnaden) [inline lib/template_requirements_config.yaml:1]: FYI - this is changing in a future version to the underscore version (Steve set me straight)

File: lib/requirements_config.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+test_kit_id: bulk-data-test-kit
+
+suites: 
+  - id: bulk_data_test_kit
```
Comment (karlnaden) [inline lib/requirements_config.yaml:4]: There are two suites defined with the following ids:
- [bulk_data_v101](https://github.com/inferno-framework/bulk-data-test-kit/blob/main/lib/bulk_data_test_kit/v1.0.1/bulk_data_test_suite.rb)
- [bulk_data_v200](https://github.com/inferno-framework/bulk-data-test-kit/blob/main/lib/bulk_data_test_kit/v2.0.0/bulk_data_test_suite.rb)

You have the test suite for the 2.0.0 version represented, but need to update the Id. Consider adding an entry for the 1.0.1 version even if we aren't planning to test it at this time.

File: lib/requirements_config.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    class_name: BulkDataTestKit::BulkDataV200::BulkDataTestSuite
+    suite_actor: Provider
```
Comment (karlnaden) [inline lib/requirements_config.yaml:6]: Actors in the bulk data requirements are Client and Server, so this suite actor needs to be one of those. These tests are testing the Server, so use that here.

File: lib/requirements_config.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+requirement_sets:
+  - id: hl7.fhir.uv.bulkdata_2.0.0_reqs
+    actor_map:
+      - {spec: Client, test_kit: Server}
```
Comment (karlnaden) [inline lib/requirements_config.yaml:11]: This actor map will cause Client actor requirements to be associated with the Server. I don't think that's what we want. In this case, there probably aren't any actor changes that you need to make, so you remove that property

**Review by karlnaden — CHANGES_REQUESTED** [review]
config file data isn't quite right. Tomorrow, I will get you some commands to try that will tell you if you've gotten it right.

File: lib/template_requirements_config.yaml (hunk 1 above)
Comment (carrils) [reply lib/template_requirements_config.yaml]: I updated the comment here to reference the underscored name.

File: lib/requirements_config.yaml (hunk 4 above)
Comment (carrils) [reply lib/requirements_config.yaml]: I got a strange error when removing the actor_map property but got around it by setting it equal to `- { spec: Server, test_kit: Server }`. 

Not sure if necessary but the error was:

  `rake aborted!
  NoMethodError: undefined method `each' for nil:NilClass
  
            req_set_config['actor_map'].each do |actor_mapping|
                                       ^^^^^
  /bulk-data-test-kit/lib/inferno_requirements_tools/tasks/collect_requirements.rb:81:in `block in parse_requirement_set'`

Comment (karlnaden) [reply lib/requirements_config.yaml]: That error is saying I didn't have safe navigation on that hash entry.  I updated the code in the base repo to `req_set_config['actor_map']&.each`, which should fix that in the future. Your fix is right for now.

**Review by karlnaden — APPROVED** [review]
Looks good - thanks for the fixes. There are some additional changes to the requirements infrastructure, but this is in-line with the current code.

**Pull request merged**